File: frontend/src/api/students.ts
export interface Student {
  id: number;
  firstName: string;
  lastName: string;
  nickname: string;
  parentName: string;
  address1: string;
  address2: string;
  city: string;
  state: string;
  postalCode: string;
  email: string;
  grade: string;
}

export async function getStudents(): Promise<Student[]> {
  const response =
    await fetch("/api/students");

  if (!response.ok) {
    throw new Error(
      "Unable to retrieve students",
    );
  }

  return response.json();
}

export async function getStudent(
  id: number,
): Promise<Student> {
  const response =
    await fetch(`/api/students/${id}`);

  if (!response.ok) {
    throw new Error(
      "Unable to retrieve student",
    );
  }

  return response.json();
}

File: model/students.go
package model

// Student represents a student record.
type Student struct {
	ID         int64  `json:"id"`
	FirstName  string `json:"firstName"`
	LastName   string `json:"lastName"`
	Nickname   string `json:"nickname"`
	ParentName string `json:"parentName"`
	Address1   string `json:"address1"`
	Address2   string `json:"address2"`
	City       string `json:"city"`
	State      string `json:"state"`
	PostalCode string `json:"postalCode"`
	Email      string `json:"email"`
	Grade      string `json:"grade"`
}
